WcaOnRails/lib/auxiliary_data_computation.rb

```diff
@@ -0,0 +1,126 @@
+      current_country = Person.current.pluck(:wca_id, :countryId).to_h
+      current_continent = Hash.new do |hash, person_id|
+        hash[person_id] = Country.c_find(current_country[person_id]).continentId
+      end
+      # Get all personal records (note: people that changed their country appear once for each country).
+      personal_records_with_event = ActiveRecord::Base.connection.execute <<-SQL
+        SELECT eventId, personId, countryId, continentId, min(#{field}) value
+        FROM #{concise_table_name}
+        WHERE eventId <> '333mbo'
+        GROUP BY personId, countryId, continentId, eventId
+        ORDER BY eventId, value
+      SQL
+      personal_records_with_event.group_by(&:first).each do |event_id, personal_records|
+        personal_rank = Hash.new { |h, k| h[k] = {} }
+        ranked = Hash.new { |h, k| h[k] = {} }
+        counter = Hash.new(0)
+        current_rank = Hash.new(0)
+        previous_value = {}
+        personal_records.each do |_, person_id, country_id, continent_id, value|
+          # Update the region states (unless we have ranked this person already,
+          # e.g. 2008SEAR01 twice in North America and World because of his two countries).
+          ["World", continent_id, country_id].each do |region|
+            next if ranked[region][person_id]
+            counter[region] += 1
+            # As we ordered by value it can either be greater or tie the previous one.
+            current_rank[region] = counter[region] if previous_value[region].nil? || value > previous_value[region]
+            previous_value[region] = value
+            ranked[region][person_id] = true
+          end
+          # Set the person's data (first time the current location is matched).
+          personal_rank[person_id][:best] ||= value
+          personal_rank[person_id][:world_rank] ||= current_rank["World"]
+          if continent_id == current_continent[person_id]
+            personal_rank[person_id][:continent_rank] ||= current_rank[continent_id]
+          end
+          if country_id == current_country[person_id]
+            personal_rank[person_id][:country_rank] ||= current_rank[country_id]
+          end
+        end
+        values = personal_rank.map do |person_id, rank_data|
+          # Note: continent_rank and country_rank may be not present because of a country change, in such case we default to 0.
+          "('#{person_id}', '#{event_id}', #{rank_data[:best]}, #{rank_data[:world_rank]}, #{rank_data[:continent_rank] || 0}, #{rank_data[:country_rank] || 0})"
+        end.join(",\n")
+        ActiveRecord::Base.connection.execute <<-SQL
+          INSERT INTO #{table_name} (personId, eventId, best, worldRank, continentRank, countryRank) VALUES
+          #{values}
+        SQL
+      end
+    end
+  end
+
+  def self.delete_php_cache
+    cache_files = Dir.glob(Rails.root.join("../webroot/results/generated/cache/*.cache"))
+    FileUtils.rm cache_files
```

Comment:
Which file do you mean? There are many files. `Rails.root.join...` returns the absolute path, `Dir.glob` turns this into a list of files matching the path (could be an empty array), and finally `FileUtils.rm` obviously gets rid of them.